adapter/service/outlvlswitch/service.go

```diff
@@ -154,12 +158,12 @@ func (s *service) SendLevelReport(force bool) (bool, error) {
 }
 
 // SetLevel sets a level value.
-func (s *service) SetLevel(value int64, duration time.Duration) error {
+func (s *service) SetLevel(value int64, duration *time.Duration) error {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	if duration > 0 {
```

Comment:
Please extract a `duration` first, so we can get rid of an `if` statement and call `SetLevelSwitchLevel` once.